### PR TITLE
Fixes #3253 Cookbook generation failure missing file indicator

### LINKED ISSUE #3253: Cookbook generation failure missing responsible file indicator
Currently cookbook generation does not provide any debug information. Therefore erroneous translation files are hard to track down.

```
Example:
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vagrant/shogun/build
[4/5] Generating examples from meta-language
FAILED: cd /home/vagrant/shogun/build/examples/meta && /usr/bin/python /home/vagrant/shogun/examples/meta/generator/generate.py -i /home/vagrant/shogun/examples/meta/src -o /home/vagrant/shogun/build/examples/meta -t /home/vagrant/shogun/examples/meta/generator/targets -g /home/vagrant/shogun/build/src/shogun/tags --store-vars
Traceback (most recent call last):
  File "/home/vagrant/shogun/examples/meta/generator/generate.py", line 120, in <module>
    includedTargets=args.targets, storeVars=storeVars)
  File "/home/vagrant/shogun/examples/meta/generator/generate.py", line 44, in translateExamples
    targets.append(json.load(tFile))
  File "/usr/lib/python2.7/json/__init__.py", line 290, in load
    **kw)
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by arianepaola**
This fixes #3253 Cookbook generation failure missing responsible file indicator.

### CODE REVIEW

**Comment by arianepaola** [comment]
@vigsterkr @karlnapf @lgoetz

Please review and merge.

**Comment by karlnapf** [comment]
I added this in f2b6e3b0970c063124ee586488995e60c20f5853

**Comment by karlnapf** [comment]
Let me know if you think this one is better, closing for now

**Comment by arianepaola** [comment]
@karlnapf The crash happens earlier, when loading json:
https://github.com/shogun-toolbox/shogun/blob/f2b6e3b0970c063124ee586488995e60c20f5853/examples/meta/generator/generate.py#L42

**Comment by karlnapf** [comment]
As discussed in irc, a try - except would be better

**Comment by arianepaola** [comment]
@karlnapf I updated this one.

File: examples/meta/generator/generate.py [hunk 1]
```diff
@@ -37,10 +37,15 @@ def translateExamples(inputDir, outputDir, targetsDir, ctagsFile,
...
         if includedTargets and not os.path.basename(target).split(".")[0] in includedTargets:
             continue
-
-        with open(os.path.join(targetsDir, target)) as tFile:
-            targets.append(json.load(tFile))
-
+        
+        translate_file = os.path.join(targetsDir, target)
+        
+        with open(translate_file) as tFile:
```
Comment (karlnapf) [inline examples/meta/generator/generate.py]: is this both Python 2 and 3 compatible?

Comment (arianepaola) [inline examples/meta/generator/generate.py]: @karlnapf yes

**Comment by karlnapf** [comment]
Good then, can you squash the commits into a single one? Then fine to merge

**Comment by arianepaola** [comment]
@karlnapf squashed in latest PR update.

**Comment by karlnapf** [comment]
Thanks

**Pull request merged**